### PR TITLE
Fix file batch operation

### PR DESCRIPTION
**Pull request by RustedBones**
RemoteFileUtil did not preserve element ordering wich was assumed in the FileDownloadDoFn

Pane information was not propagated

### CODE REVIEW

File: scio-core/src/main/java/com/spotify/scio/transforms/FileDownloadDoFn.java [hunk 1]
```diff
@@ -103,51 +123,31 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
-      return;
-    }
-    LOG.info("Processing batch of {}", batch.size());
-    List<URI> uris = batch.stream().map(e -> e.uri).collect(Collectors.toList());
-    List<OutputT> outputs =
-        remoteFileUtil.download(uris).stream().map(fn::apply).collect(Collectors.toList());
-    // .forEach(c::output);
-    Iterator<OutputT> i1 = outputs.iterator();
-    Iterator<Element> i2 = batch.iterator();
-    while (i1.hasNext() && i2.hasNext()) {
```
Comment (RustedBones) [inline scio-core/src/main/java/com/spotify/scio/transforms/FileDownloadDoFn.java:131]: This parts assumes that `List<Output>` is 1:1 mapping of  `List<URI>`

File: scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java [hunk 2]
```diff
@@ -104,7 +106,12 @@ public Path download(URI src) {
    */
   public List<Path> download(List<URI> srcs) {
     try {
-      return paths.getAll(srcs).values().asList();
```
Comment (RustedBones) [inline scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java:107]: `getAll(srcs)` returns a `Map<URI, Path>`. Using `values` give no guarantee that returned `Paths` are ordered.

File: scio-core/src/main/java/com/spotify/scio/transforms/FileDownloadDoFn.java [hunk 3]
```diff
@@ -103,51 +123,31 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
   }
 
-  private void processBatch(OutputReceiver<OutputT> outputReceiver) {
+  private void flush(Consumer<ValueInSingleWindow<OutputT>> outputFn) {
     if (batch.isEmpty()) {
       return;
     }
     LOG.info("Processing batch of {}", batch.size());
-    List<URI> uris = batch.stream().map(e -> e.uri).collect(Collectors.toList());
-    remoteFileUtil.download(uris).stream().map(fn::apply).forEach(outputReceiver::output);
```
Comment (RustedBones) [inline scio-core/src/main/java/com/spotify/scio/transforms/FileDownloadDoFn.java:112]: original element metadata are not passed back

File: scio-core/src/test/scala/com/spotify/scio/transforms/FileDownloadDoFnTest.scala [hunk 4]
```diff
@@ -42,9 +47,23 @@ class FileDownloadDoFnTest extends PipelineSpec {
...
+        .timestampBy { case (_, i) => new Instant(i + 1) }
+        .keys
+        .flatMapFile(fn, 10, false)
+        .withTimestamp
+
+      val contentAndTimestamp = p.map { case ((i, _), ts) => (i, ts.getMillis) }
+      val paths =  p.map { case ((_, f), _) => f }.distinct
+
+      val expected = (1L to 100L).map(i => (i.toString, i))
+      contentAndTimestamp should containInAnyOrder(expected)
```
Comment (RustedBones) [inline scio-core/src/test/scala/com/spotify/scio/transforms/FileDownloadDoFnTest.scala:65]: Ran on the original code, got non expected item `(1, 10)` -> (1st element was flushed while processing the 10th)

**Review by kellen — APPROVED** [review]
Nice catch!

**Pull request merged**